### ISSUE TITLE
YouTube Music Playlists don't show up

### ISSUE THREAD

**Issue by rjgrandy** [music-assistant/hass-music-assistant#663]
### What version of Music Assistant has the issue?

2022.7.b5

### The problem

When I navigate to the playlists section under youtube music, the playlists never load, it shows the loading bar forever and I get the following error in the logs:

```2022-07-14 11:24:57 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140717956706512] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/mass/websockets.py", line 106, in async_get_mass_func
    await orig_func(hass, connection, msg, mass)
  File "/config/custom_components/mass/websockets.py", line 856, in websocket_browse
    result = await mass.music.browse(msg.get(URI))
  File "/usr/local/lib/python3.10/site-packages/music_assistant/controllers/music/__init__.py", line 209, in browse
    return await prov.browse(path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 273, in browse
    return [x async for x in self.get_library_playlists()]
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 273, in 
    return [x async for x in self.get_library_playlists()]
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/ytmusic.py", line 134, in get_library_playlists
    playlists_obj = await get_library_playlists(headers=self._headers)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 125, in get_library_playlists
    return await loop.run_in_executor(None, _get_library_playlists)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 117, in _get_library_playlists
    playlists = ytm.get_library_playlists(limit=9999)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/mixins/library.py", line 32, in get_library_playlists
    playlists = parse_content_list(results['items'][1:], parse_playlist)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 158, in parse_content_list
    search_results.append(search_result)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 196, in parse_playlist
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/utils.py", line 155, in nav
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/utils.py", line 149, in nav
KeyError: 'runs'
```

### How to reproduce

Navigate to the playlist section under youtube music

### Relevant log output

2022-07-14 11:24:57 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140717956706512] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/mass/websockets.py", line 106, in async_get_mass_func
    await orig_func(hass, connection, msg, mass)
  File "/config/custom_components/mass/websockets.py", line 856, in websocket_browse
    result = await mass.music.browse(msg.get(URI))
  File "/usr/local/lib/python3.10/site-packages/music_assistant/controllers/music/__init__.py", line 209, in browse
    return await prov.browse(path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 273, in browse
    return [x async for x in self.get_library_playlists()]
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 273, in 
    return [x async for x in self.get_library_playlists()]
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/ytmusic.py", line 134, in get_library_playlists
    playlists_obj = await get_library_playlists(headers=self._headers)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 125, in get_library_playlists
    return await loop.run_in_executor(None, _get_library_playlists)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 117, in _get_library_playlists
    playlists = ytm.get_library_playlists(limit=9999)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/mixins/library.py", line 32, in get_library_playlists
    playlists = parse_content_list(results['items'][1:], parse_playlist)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 158, in parse_content_list
    search_results.append(search_result)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 196, in parse_playlist
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/utils.py", line 155, in nav
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/utils.py", line 149, in nav
KeyError: 'runs'

### What version of Home Assistant Core are your running

2022.7.0

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Linux

**Comment by MarvinSchenkel** [comment]
So I think I figured out what is wrong. Looking at the log, you are using a (very) old version of an underlying ytmusicapi library that is not supported with Music Assistant. You probably have installed the other YT Music custom component by KoljaWindeler, which for *some* reason causes a library collision.

This shouldn't be possible, so I am wondering whether you installed the component manually or via HACS?

To fix this issue:
- Uninstall Youtube Music Player
- Uninstall Music Assistant
- Reboot HA
- Install Music Assistant via HACS

**Comment by rjgrandy** [comment]
Thanks for the response, however , it doesn't seem to be working still after removing all traces of the previous ytmusicapi from HA. Now when I click on "Browse" then "YouTube Music", nothing happens, it just stays on the same screen. 
When I click "Playlists" from the main screen nothing happens. The other categories are populated with items though. 

I also don't get anymore error logs. I tried to change the logging to debug but it doesn't show anything else for when I try to go into playlists.

Is there any logs that I can capture that might help?

**Comment by MarvinSchenkel** [comment]
> Thanks for the response, however , it doesn't seem to be working still after removing all traces of the previous ytmusicapi from HA. Now when I click on "Browse" then "YouTube Music", nothing happens, it just stays on the same screen. When I click "Playlists" from the main screen it says "No items found.". The other categories are populated with items though.
> 
> I also don't get anymore error logs. I tried to change the logging to debug but it doesn't show anything else for when I try to go into playlists.
> 
> ```
> service: logger.set_level
> data:
>   music_assistant: debug
> ```
> 
> Is there any logs that I can capture that might help?

The ID of Music Asssitant is `mass`, so you would have to call it like this to get the debug logs:
```yaml
service: logger.set_level
data:
  custom_components.mass: debug
```

or update the `logger` section of your `configuration.yaml`:

```yaml
---
logger:
  default: error
  logs:
    custom_components.mass: debug
```

One more question, did you install Music Assistant via HACS?

**Comment by rjgrandy** [comment]
Yes I installed Music Assistant via HACS.

I may have originally installed YT Music custom component outside of HACS and then later added via HACS. However I have removed all the files related to the YT Music custom component I can find.

I don't see any additional logs related to the playlists (only additional logs after clicking an album or artist) after changing the logger.set_level as you show above.

**Comment by erkr** [comment]
@rjgrandy 
Can you please check if this issue still persists in the latest beta 8? Browsing YS playlist works fine for me.

**Comment by rjgrandy** [comment]
@erkr 
Yes I still get the same error when I go to the browse - playlists area. I do have another error when I do a force full sync, perhaps that would help?

The 2 errors are:

**Browse - Playlists**
```
2022-07-21 11:00:47 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139644981034336] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/mass/websockets.py", line 111, in async_get_mass_func
    await orig_func(hass, connection, msg, mass)
  File "/config/custom_components/mass/websockets.py", line 894, in websocket_browse
    await mass.music.browse(msg.get("path")),
  File "/usr/local/lib/python3.10/site-packages/music_assistant/controllers/music/__init__.py", line 218, in browse
    return await prov.browse(path)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 370, in browse
    items=[x async for x in self.get_library_playlists()],
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 370, in 
    items=[x async for x in self.get_library_playlists()],
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/ytmusic.py", line 154, in get_library_playlists
    playlists_obj = await get_library_playlists(
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 135, in get_library_playlists
    return await loop.run_in_executor(None, _get_library_playlists)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 126, in _get_library_playlists
    playlists = ytm.get_library_playlists(limit=9999)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/mixins/library.py", line 32, in get_library_playlists
    playlists = parse_content_list(results['items'][1:], parse_playlist)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 195, in parse_content_list
    contents.append(parse_func(result[key]))
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 267, in parse_playlist
    'title': nav(data, TITLE_TEXT),
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/navigation.py", line 72, in nav
    raise err
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/navigation.py", line 66, in nav
    root = root[k]
KeyError: 'runs'
```
**Force Full Sync**
```
2022-07-21 11:01:42 ERROR (MainThread) [music_assistant] Job [Library sync for provider Youtube Music] failed with error 'runs'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/music_provider.py", line 398, in sync_library
    async for prov_item in self._get_library_gen(media_type)():
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/ytmusic.py", line 154, in get_library_playlists
    playlists_obj = await get_library_playlists(
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 135, in get_library_playlists
    return await loop.run_in_executor(None, _get_library_playlists)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/music_providers/ytmusic/helpers.py", line 126, in _get_library_playlists
    playlists = ytm.get_library_playlists(limit=9999)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/mixins/library.py", line 32, in get_library_playlists
    playlists = parse_content_list(results['items'][1:], parse_playlist)
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 195, in parse_content_list
    contents.append(parse_func(result[key]))
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/parsers/browsing.py", line 267, in parse_playlist
    'title': nav(data, TITLE_TEXT),
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/navigation.py", line 72, in nav
    raise err
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/navigation.py", line 66, in nav
    root = root[k]
KeyError: 'runs'
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/navigation.py", line 72, in nav
    raise err
  File "/usr/local/lib/python3.10/site-packages/ytmusicapi/navigation.py", line 66, in nav
    root = root[k]
KeyError: 'runs'
```

**Comment by erkr** [comment]
Thanks, we keep this issue open then!

**Comment by rjgrandy** [comment]
@MarvinSchenkel Is there any other things I can do or provide to help debug this?

**Comment by MarvinSchenkel** [comment]
It looks like you have an exotic playlist in your library. Looking at the logs you provided, it looks like you have a playlist without a title in your library. Could you maybe go to https://music.youtube.com/library/playlists and see if you see any strange playlists? If not, please include a screenshot of your playlists so I can have a look as well.

**Comment by rjgrandy** [comment]
That was it! There was an unnamed Playlist with 0 items. Once I deleted it, the Playlist started showing up. Thanks

**Comment by txwindsurfer** [comment]
@MarvinSchenkel I apologize in advance for posting to a closed issue but my issue is so similar that I hope this makes it a little easier.  I do not have an exotic name in my playlist, but I cannot get the playlist for YTM to load in Music Assistant. I am running version 2022.8.1 of MA and 22.8.2 of HA. Configuration is correct I think because if I go to the Media tab and select a player (output device) I can see Music Assistant as an option for the input.  When I select MA as the input I can see the playlist, albums, artists, etc from YTM in Media.  However, nothing will show in MA and the playlist, album, artist is grayed out.  I came to this issue because I also had previously installed the other YT Music custom component by KoljaWindeler, which I removed following the procedure above in this thread.  Perhaps there is still a library I need to remove.  Any help would be very appreciated and again I apologize if I should not have posted into a closed issue.
